### ISSUE TITLE
can we have a place to pull request presets?

### ISSUE THREAD

**Issue by RustoMCSpit** [surge-synthesizer/surge#7622]
i think having a place to PR our presets for everyone, like the skins you provide, would be very helpful. having a large community providing presets which people can choose to download for free would be great (and if theyre good enough, be included in the factory bank)

**Comment by mkruselj** [comment]
We have a manual screening process for admitting presets to the factory set, and this is all happening on Discord. Pull requesting things willy nilly is not likely going to happen.

**Comment by RustoMCSpit** [comment]
okay then forget about the factory set. can we have a place, maybe in a different repo under surge, where users can freely PR their presets. in fairness i feel like this should be on the website so people can vote and sort by tag

**Comment by RustoMCSpit** [comment]
tl;dr - add a place for users to upload presets for everyone for free (i know theres third parties but standardisation is key for adoption and it's so much simpler)

**Comment by baconpaul** [comment]
Why not just try to set up a surge community in https://patchstorage.com/explore/? Works for rack well and some other platforms

**Comment by RustoMCSpit** [comment]
> Why not just try to set up a surge community in https://patchstorage.com/explore/? Works for rack well and some other platforms

ive never seen this site. the thing is i have no power in this matter, i dont have sway over what the WHOLE community does. but if there was some official means in the readme which everyone knew about, even just a separate repo where people take PRs from (or something better, or something like your website or teaming up with tonehunt.org which is also open source) thatd work a lot better

does that all make sense?

**Comment by Andreya-Autumn** [comment]
In a sense yes. But remember, this is an open-contribution project. Which means you have just as much as power as Paul or Mario or I do to set up a patchstorage page! Though it's of course a good idea to communicate about such an endeavor with the team before doing it, which you are doing.

I'd be happy to link to such a patchstorage page from the website/readme/discord/kvr thread if you did. So if you feel like that'd be a useful thing to exist in the world, by all means go ahead. :)

**Comment by RustoMCSpit** [comment]
is patchstorage an ideal place to do this, again im not familiar with the service. is it open source? will it ever start charging payments? also, i didnt know you had a discord but have you tried revolt (foss discord clone)) https://github.com/revoltchat

**Comment by Andreya-Autumn** [comment]
Yeah Discord is where we handle most of our internal communications right now (and bug reports from folks who don't use Github). Thanks for the tip about Revolt! We'd definitely consider it if at some point Discord doesn't work for us anymore. 

Patchstorage is a well-established platform with a good reputation, though I have not used it myself and don't know much more than that. It seems like the right place for what you're proposing thematically. 
Whereas Tonehunt with its focus on guitar amp/effect models and IRs and such probaby isn't the right platform for Surge.  

The reason we're reluctant to manage a custom patch library platform ourselves is simply that it's more work. And all the currently active team members have plenty other things going on (like developing and shipping Surge 1.3.2, continuing to improve Shortcircuit, plus managing our already active platform on Discord etc etc), which I'm guessing we don't wanna divert from. 

If someone shows up willing to commit to the work of setting such a platform up and managing it, sure! But definitely the easiest way to start something up would be to use an existing platform like patchstorage. Hence the rec.

**Comment by RustoMCSpit** [comment]
> Whereas Tonehunt with its focus on guitar amp/effect models and IRs and such probaby isn't the right platform for Surge.

https://github.com/tonehunt-org/tonehunt/issues/269 theyre interested in increasing their range and im just trying to think long-run that using another open source project would be best but ik what you mean

> The reason we're reluctant to manage a custom patch library platform ourselves is simply that it's more work
the hope is that it's a platform that self-regulates through things like ratings, comments, etc.. the best automatically will reach the top

>  to use an existing platform like patchstorage. Hence the rec.
ask around discord for alternatives

**Comment by Andreya-Autumn** [comment]
https://patchstorage.com/platform/vcv-rack/

From the looks of this it seems like this is exactly what you're looking for, no? Why the skepticism (if that is indeed skepticism I perceive)?

**Comment by RustoMCSpit** [comment]
> https://patchstorage.com/platform/vcv-rack/
> 
> From the looks of this it seems like this is exactly what you're looking for, no? Why the skepticism (if that is indeed skepticism I perceive)?

not neccessarily skepticism, i just dont know if theres better alternatives. if theres an open source alternative, it'd be better though

**Comment by mkruselj** [comment]
Maybe, maybe not, you cannot say for sure.

At any rate, moving forward and especially for Surge XT2, whenever it happens, there will be a more stringent process for ingesting patches into factory bank and the emphasis will be on quality over quantity (for example: all macros will have to be defined and do meaningful things in ever patch, which currently is not the case at all). Hence these automatic ways of patch submission with no semblance of quality control are in direct opposition to what we have planned.

**Comment by RustoMCSpit** [comment]
> Maybe, maybe not, you cannot say for sure.
> 
> At any rate, moving forward and especially for Surge XT2, whenever it happens, there will be a more stringent process for ingesting patches into factory bank and the emphasis will be on quality over quantity (for example: all macros will have to be defined and do meaningful things in every patch, which is currently not the case at all). Hence these automatic ways of patch submission with no semblance of quality control are in direct opposition to what we have planned.

no wait, forget about the factory bank. im just looking for a standardised, official way that users can share patches. if guidelines like meaningful macros are a must thats fine

**Comment by mkruselj** [comment]
Anyone is welcome to our Discord and #surge-content-sharing channel.

**Comment by RustoMCSpit** [comment]
> surge-content-sharing channel.

yeah something like that but on an open source platform a bit more tailor made for stuff like this.

**Comment by RustoMCSpit** [comment]
does that make sense

**Comment by mkruselj** [comment]
In that case I defer to everything Andreya already said. We as Surge Synth Team will not be creating such a platform. We have enough things to maintain as is.

**Comment by RustoMCSpit** [comment]
> In that case I defer to everything Andreya already said. We as Surge Synth Team will not be creating such a platform as we already have enough things to maintain as is.

no i was wondering if there was already a platform which an account could be made on and then labelled as the official place for surge presets

**Comment by mkruselj** [comment]
My own stance is that only what comes installed with Surge is official.

**Comment by RustoMCSpit** [comment]
> My own stance is that only what comes installed with Surge is official.

im not asking for changes to the factory bank, i am asking for an official place to publish presets. if thats on a third party service, great. the discord isnt suited for this task. i am not askjng for mods to review all the presets, just set basic guidelines like 'are the macros properly named' 'are the presets uniquely named' 'are they properly tagged?'

**Comment by Andreya-Autumn** [comment]
That's understood! Here's the thing, what I gather is that you A: have clear opinions on how this should be done, and B: want one of us to do it. 

If I'm hearing you wrong, let me know of course. But basically A and B do not combine well. I know you don't mean to be rude! But It comes across a bit like "you should do this, but you should do it my way", which no one wants to hear in a volunteer project. 

But I would suggest the following: 
1: We agree that patchstorage is a good option, in which case let's proceed from there. 
Or 2: You don't feel it's good enough, in which case can you please do the research of finding a better alternative and present it here? 

I think you're right that such a place could be a good idea. As Paul said, it works well for the VCV community. So if we can agree to something I don't oppose it! It's just the "assigning of work" to others that kinda doesn't sit right with this project. See what I mean?

**Comment by Andreya-Autumn** [comment]
Right! Wanna set up a patch storage page then? Ideally with a username/email/password that you could pass on to someone else in the future in the event that becomes useful? I can set up links from relevant places once it exists. 

Make sure you name it Surge XT if you do, to avoid ambiguity with the original (pre-port) Surge.

**Comment by RustoMCSpit** [comment]
> Right! Wanna set up a patch storage page then? Ideally with a username/email/password that you could pass on to someone else in the future in the event that becomes useful? I can set up links from relevant places once it exists.
> 
> Make sure you name it Surge XT if you do, to avoid ambiguity with the original (pre-port) Surge.

before i do that, could you post to the discord if anyone has any better suggestions? will do that if no one has any better ideas? again, an open source alternative would be ideal. how large are surge xt presets typically?

**Comment by RustoMCSpit** [comment]
ive asked the tonehunt team if theyll consider this, if so, i'll hold off on the non-open source option

**Comment by RustoMCSpit** [comment]
> ive asked the tonehunt team if theyll consider this, if so, i'll hold off on the non-open source option

dev response:
Hi! Im not familiar with SurgeXT, but I'll check it out. Looks interesting. 
Regarding hosting surgext presets on TH: we're currently on the final stages of our next release, which will start bringing support to other platforms like Aida-X and Proteus. Once we're done with this release, we could take a look to see if it makes sense to have surgext stuff here and what would it take dev-wise. We do have a lot in our pipeline (which we'll publish after the release) so this could take a while to even look at (just to set some expectations). But we'll do our best and we'll give an update later. Thanks for reaching out!

**Comment by Andreya-Autumn** [comment]
> before i do that, could you post to the discord if anyone has any better suggestions? will do that if no one has any better
ideas? again, an open source alternative would be ideal.

"Can you please do the research you just asked me to do?" 

Listen, this won't work here. 

In your defense you did reach out to that other project. Appreciate that. But if I'm honest, I doubt that it would be the right platform for this. 

So again (and probably for the last time). Patch storage seems like the right answer to us. Appreciate the open source perspective (obviously), but this project also relies on a "let's get things done effectively" perspective.

**Comment by RustoMCSpit** [comment]
> "Can you please do the research you just asked me to do?"
Listen, this won't work here.

sorry thats fair criticism.

i'll look into patch storage

**Comment by RustoMCSpit** [comment]
update studio rack dev says theyre thinking on working on this! still working out kinks and no guarantees

**Comment by baconpaul** [comment]
they seem to have a 3 year old version of surge in their repo, if you are talking to them. never heard of the project though.

**Comment by RustoMCSpit** [comment]
> they seem to have a 3 year old version of surge in their repo, if you are talking to them. never heard of the project though.

it's in beta

**Comment by RustoMCSpit** [comment]
> update studio rack dev says theyre thinking on working on this! still working out kinks and no guarantees

the project has evolved since then. it's nearly ready to start hosting presets. https://github.com/open-audio-stack/open-audio-stack-registry

**Comment by RustoMCSpit** [comment]
> they seem to have a 3 year old version of surge in their repo, if you are talking to them. never heard of the project though.

it's updated

**Comment by baconpaul** [comment]
it just doesn't work

the mac binary you use is the plugins only unsigned zip as opposed to the installer so the download gives you an incorrectly configured non runnable surge.

just fyi

**Comment by baconpaul** [comment]
I dropped a dm to ktumurly with some more details.

**Comment by RustoMCSpit** [comment]
> it just doesn't work
> 
> the mac binary you use is the plugins only unsigned zip as opposed to the installer so the download gives you an incorrectly configured non runnable surge.
> 
> just fyi

once this is all resolved, which is already under works, i'll ping again. still some issues as a new registry was just implemented four days ago. thanks!!

**Comment by baconpaul** [comment]
It’s already fixed!

**Comment by RustoMCSpit** [comment]
> It’s already fixed!

oh lovely, so is there anything else needed before this is made an official site? what would be the to do list?

**Comment by baconpaul** [comment]
ktmurley is still doing some background work but yeah he and i are chatting

i think the only thing to do is link to his effort from somewhere right? its not like we're going to actually do anything other than chat with the team and inform you all when we do upgrades and so forth. (and i'm thinking maybe convert to msi(

**Comment by RustoMCSpit** [comment]
so all you want is basically an rss feed

**Comment by RustoMCSpit** [comment]
> its not like we're going to actually do anything other than chat with the team and inform you all when we do upgrades

https://github.com/open-audio-stack/open-audio-stack-registry/issues/24

when this is done all devs will be handed an account and passcode so you can update all of your plugins, set up donation links, etc.

**Comment by RustoMCSpit** [comment]
> ktmurley is still doing some background work but yeah he and i are chatting
> 
> i think the only thing to do is link to his effort from somewhere right? its not like we're going to actually do anything other than chat with the team and inform you all when we do upgrades and so forth. (and i'm thinking maybe convert to msi(

Please do that, linking it in the readme should be fine. OpenAudio is now past alpha and it's being hosted on StudioRack and OwlPlug https://github.com/DropSnorz/OwlPlug

**Comment by RustoMCSpit** [comment]
@kmturley

**Comment by baconpaul** [comment]
ok maybe i'll take 30 minutes this weekend and add six sines to your repo and see what happens

it bums me out a lot that owl plug starts with directions on how to turn off mac security to use it. are there plans to properly sign the asset? or is that a separate project.

**Comment by RustoMCSpit** [comment]
OwlPlug is third party, StudioRack is soon to implement OAS https://github.com/studiorack/studiorack-app however it has implemented it on its website https://studiorack.github.io/studiorack-site

**Comment by RustoMCSpit** [comment]
> ok maybe i'll take 30 minutes this weekend and add six sines to your repo and see what happens
> 
> it bums me out a lot that owl plug starts with directions on how to turn off mac security to use it. are there plans to properly sign the asset? or is that a separate project.

@dropsnorz

**Comment by baconpaul** [comment]
Ok well six sines pull request is in

**Comment by RustoMCSpit** [comment]
https://open-audio-stack.github.io/open-audio-stack-site

Hi all, just wanted to update. It is now far easier to upload presets. Please give it a try! If you think this is good enough to be an officially endorsed place for preset uploads, please it add to your README. Thanks!
